src/test/java/com/shapesecurity/shift/parser/RangeCheckerReducer.java

```diff
@@ -0,0 +1,711 @@
+	}
+
+	@NotNull
+	@Override
+	public RangeChecker reduceTryCatchStatement(
+		@NotNull TryCatchStatement node, @NotNull RangeChecker block, @NotNull RangeChecker catchClause
+	) {
+		return accept(node, super.reduceTryCatchStatement(node, block, catchClause));
+	}
+
+	@NotNull
+	@Override
+	public RangeChecker reduceTryFinallyStatement(
+		@NotNull TryFinallyStatement node, @NotNull RangeChecker block, @NotNull Maybe<RangeChecker> catchClause,
+		@NotNull RangeChecker finalizer
+	) {
+		return accept(node, super.reduceTryFinallyStatement(node, block, catchClause, finalizer));
+	}
+
+	@NotNull
+	@Override
+	public RangeChecker reduceUnaryExpression(@NotNull UnaryExpression node, @NotNull RangeChecker operand) {
+		return accept(node, super.reduceUnaryExpression(node, operand));
+	}
+
+	@NotNull
+	@Override
+	public RangeChecker reduceUpdateExpression(@NotNull UpdateExpression node, @NotNull RangeChecker operand) {
+		return accept(node, super.reduceUpdateExpression(node, operand));
+	}
+
+	@NotNull
+	@Override
+	public RangeChecker reduceVariableDeclaration(
+		@NotNull VariableDeclaration node, @NotNull ImmutableList<RangeChecker> declarators
+	) {
+		return accept(node, super.reduceVariableDeclaration(node, declarators));
+	}
+
+	@NotNull
+	@Override
+	public RangeChecker reduceVariableDeclarationStatement(
+		@NotNull VariableDeclarationStatement node, @NotNull RangeChecker declaration
+	) {
+		return accept(node, super.reduceVariableDeclarationStatement(node, declaration));
+	}
+
+	@NotNull
+	@Override
+	public RangeChecker reduceVariableDeclarator(
+		@NotNull VariableDeclarator node, @NotNull RangeChecker binding, @NotNull Maybe<RangeChecker> init
+	) {
+		return accept(node, super.reduceVariableDeclarator(node, binding, init));
+	}
+
+	@NotNull
+	@Override
+	public RangeChecker reduceWhileStatement(
+		@NotNull WhileStatement node, @NotNull RangeChecker test, @NotNull RangeChecker body
+	) {
+		return accept(node, super.reduceWhileStatement(node, test, body));
+	}
+
+	@NotNull
+	@Override
+	public RangeChecker reduceWithStatement(
+		@NotNull WithStatement node, @NotNull RangeChecker object, @NotNull RangeChecker body
+	) {
+		return accept(node, super.reduceWithStatement(node, object, body));
+	}
+
+	@NotNull
+	@Override
+	public RangeChecker reduceYieldExpression(@NotNull YieldExpression node, @NotNull Maybe<RangeChecker> expression) {
+		return accept(node, super.reduceYieldExpression(node, expression));
+	}
+
+	@NotNull
+	@Override
+	public RangeChecker reduceYieldGeneratorExpression(
+		@NotNull YieldGeneratorExpression node, @NotNull RangeChecker expression
+	) {
+		return accept(node, super.reduceYieldGeneratorExpression(node, expression));
+	}
+
+	static class RangeChecker {
+		public final static Monoid<RangeChecker> MONOID = new Monoid<RangeChecker>() {
+			@NotNull
+			@Override
+			public RangeChecker identity() {
+				return new RangeChecker(Integer.MAX_VALUE, Integer.MIN_VALUE);
```

Comment:
It's the identity, so it should compare against any other range favourably. When compared "on the right", it should always start *after* the end of the other range. Similarly, when compared "on the left", it should always end *before* the beginning of the other range. So these numbers are right.